lib/Ast.ml

```diff
@@ -1644,7 +1644,8 @@ end = struct
           match i.[0] with
           | '!' | '?' | '~' -> Some (High, Non)
           | _ -> Some (Apply, Non) ) )
-      | Pexp_apply ({pexp_desc= Pexp_ident ident; _}, args)
+      | Pexp_apply
+          ({pexp_desc= Pexp_ident ident; pexp_attributes= []; _}, args)
         when Option.is_some (Indexing_op.get_sugar ident args) -> (
```

Comment:
Just to make sure, is there still a test that covers this branch now?

Reply:
Yes this is the default case, when there is `String.get s i` that will automatically get sugared into the bracket form, we have some tests already.

Reply:
allright!

Reply:
Perhaps we should directly pass the first argument of `Pexp_apply` to `get_sugar` and move such details where they make sense ?

Reply:
Good idea, done!